code/modules/mining/minebot.dm

```diff
@@ -94,7 +94,8 @@
 		qdel(action)
 
 	// Clear any equipment they might have
-	QDEL_LIST(installed_upgrades)
+	for(var/upgrade in installed_upgrades)
```

Comment:
Keep QDEL_LIST here

Reply:
QDEL_LIST runtimes in this case, since it tries to cut a null list after everything's been removed.

Reply:
if the list is null the macro doesn't do anything, QDEL_LIST without the for as was originally should be perfectly fine

Reply:
Null.cut() runtimes, as it has a couple times on master before this change (which I didn't notice while testing my original PR).

Reply:
To be more specific, this is probably happening because I've used lazyadds and lazyremoves to handle the equipment list, which nulls the list while it's qdeling everything inside.

Reply:
Ah, if its a lazylist that makes sense, use QDEL_LAZYLIST instead then

Reply:
TIL that's a macro. I'll fix it when I'm back at my computer.

Reply:
QDEL_LIST could be made to handle cases like this quite easily, QDEL_LAZYLIST really has not much for a reason to exist honestly.

Reply:
Alright, just did that instead. QDEL_LAZYLIST's redundant now, though, so I can just remove that if you want.